nomad/testing.go

```diff
@@ -91,8 +92,9 @@ func TestServer(t testing.T, cb func(*Config)) *Server {
 	catalog := consul.NewMockCatalog(config.Logger)
 
 	for i := 10; i >= 0; i-- {
-		// Get random ports
-		ports := freeport.GetT(t, 2)
+		// Get random ports, need to cleanup later
+		ports := freeport.MustTake(2)
```

Comment:
For Nomad ports could we use port 0 and somehow expose the port from the created listener? Just seems like such a waste to do all of this heavyweight coordination and accounting when the OS can do it for us. Or perhaps we could just allow injecting `net.Conn`s for tests?

These ideas don't need to block this PR. I just feel like there's a solution to most of our use of freeport that uses less code and coordination.

Reply:
Yeah letting the OS manage all this would be _great_, the open question of course being what to do about the associated plumbing. Gut feeling is that since our `Config` objects export `*net.TCPAddr`, things will stay icky unless maybe we replace those with some kind of port `promise`-y object